ddtrace/internal/flare/flare.py

```diff
@@ -96,35 +109,33 @@ def send(self, flare_send_req: FlareSendRequest):
                 log.error("Failed to create %s file", lock_path)
                 raise e
             try:
-                client = get_connection(config._trace_agent_url, timeout=self.timeout)
+                client = get_connection(self.url, timeout=self.timeout)
                 headers, body = self._generate_payload(flare_send_req.__dict__)
                 client.request("POST", TRACER_FLARE_ENDPOINT, body, headers)
                 response = client.getresponse()
                 if response.status == 200:
                     log.info("Successfully sent the flare to Zendesk ticket %s", flare_send_req.case_id)
                 else:
-                    log.error(
-                        "Tracer flare upload to Zendesk ticket %s failed with %s status code:(%s) %s",
-                        flare_send_req.case_id,
+                    msg = "Tracer flare upload responded with status code %s:(%s) %s" % (
                         response.status,
                         response.reason,
                         response.read().decode(),
                     )
+                    raise TracerFlareSendError(msg)
             except Exception as e:
-                log.error("Failed to send tracer flare to Zendesk ticket %s", flare_send_req.case_id)
+                log.error("Failed to send tracer flare to Zendesk ticket %s: %s", flare_send_req.case_id, e)
                 raise e
             finally:
                 client.close()
                 # Clean up files regardless of success/failure
                 self.clean_up_files()
-                return
 
-    def _generate_config_file(self, pid: int):
+    def _generate_config_file(self, config: dict, pid: int):
```

Comment:
@brettlangdon You previously mentioned having some kind of validation method for the config. Is it safe to assume here that the main keys in the `Config` object won't change much, so I could safely write a more "rigid" validation method (ex - make a static list of the top level keys, verify they exist in the JSON)? Or, are we just looking for possible human unfriendly rendering of the file (ex - make sure none of the values are just object references)?

Reply:
this is the library's config (`ddtrace.config` thing)?

or is this something we receive from remote config?

Reply:
Yep the `ddtrace.config`